src/session/src/session_config.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+use common_macro::stack_trace_debug;
+use snafu::{Location, Snafu};
+use sql::ast::Value;
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum SessionConfigOption {
+    Mysql,
+    Postgres(PostgresOption),
+}
+#[derive(Snafu)]
+#[snafu(visibility(pub))]
+#[stack_trace_debug]
+pub enum Error {
+    #[snafu(display("Unkonw Postgres option '{}'", name))]
+    UnknownOption { name: String, location: Location },
+    #[snafu(display("Invalid value: {} for '{}'", value, name))]
+    InvalidConfigValue {
+        name: String,
+        value: String,
+        location: Location,
+    },
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum PostgresOption {
+    ByteaOutput,
+}
+
+impl Display for PostgresOption {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PostgresOption::ByteaOutput => write!(f, "BYTEA_OUTPUT"),
+        }
+    }
+}
+
+impl TryFrom<&str> for PostgresOption {
+    type Error = Error;
+    fn try_from(name: &str) -> Result<Self, Self::Error> {
+        match name.to_uppercase().as_str() {
+            "BYTEA_OUTPUT" => Ok(PostgresOption::ByteaOutput),
+            _ => UnknownOptionSnafu { name }.fail(),
+        }
+    }
+}
+
+pub fn postgres_option(option: PostgresOption) -> SessionConfigOption {
+    SessionConfigOption::Postgres(option)
+}
+
+#[derive(Debug, Clone)]
+pub enum SessionConfigValue {
+    MySql,
+    Postgres(PostgresConfigValue),
+}
+
+pub fn postgres_config_value(value: PostgresConfigValue) -> SessionConfigValue {
+    SessionConfigValue::Postgres(value)
+}
+
+#[derive(Debug, Clone)]
+pub enum PostgresConfigValue {
+    ByteaOutput(ByteaOutputValue),
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum ByteaOutputValue {
+    DEFAULT,
```

Comment:
There're only two bytea output configuration in pg, and the default is "HEX". So I suggest remove the "DEFAULT" here, and make "HEX" the default by annotating it "`[default]`".

Reply:
Although pg only take two bytea, the default output of *greptime* is not same as `hex`. The hex format has a extra prefix `\x`

Reply:
I think we can, it's best to comply with pg. WDYT @sunng87

Reply:
Yes, we need to adopt the postgres way for hex.